### ISSUE TITLE
Update & review docs of this library.

### ISSUE THREAD

**Issue by CPerezz** [dusk-network/rusk-vm#72]
This has been open sourced for like a month, and we don't have published docs anywhere. 
This was also stated in #30.

Apart from that, there are a lot of `pub` exported fields & fn that are lacking documentation.

I think that this would also be nice to do as a "review" of the actual status of the library. So we can then check for errors or things that are missing or do not make sense.

This is linked to #65

**Comment by CPerezz** [comment]
@krl mentioned that this repo was no longer compiling due to some problems w/ `phoenix-abi`.

Could you please open a separate issue for it?

**Comment by krl** [comment]
This is blocked by refactor of phoenix with new hades

**Comment by autholykos** [comment]
Closing as obsolete

**Comment by autholykos** [comment]
Closing as obsolete
